modules/javafx.graphics/src/main/java/com/sun/javafx/scene/paint/TextureData.java

```diff
@@ -0,0 +1,110 @@
+package com.sun.javafx.scene.paint;
+
+// When the public API is ready, move to the exported javafx.scene.paint
+/**
+ * Contains texture properties for a {@code Material}'s textures (maps).
+ * <p>
+ * Since {@code TextureData} does not depend on a {@code Material}, the same instance can be reused in multiple textures
+ * (in the same or in different materials). This is useful because many times a {@link javafx.scene.shape.Shape3D} will
+ * have all its textures use the same properties, or the same texture data will be used across the whole application.
+ */
+// Here we can support mipmaps, wrapping modes (which exists internally and can be pulled out), addressing modes etc.
+public class TextureData {
```

Comment:
1. This type could be a record.
2. There's no actual texture data (i.e. pixels) here, maybe a better name would be `SamplerParameters` or something like that.

Reply:
1. Not when it will be promoted to public API. Adding record components breaks backwards compatibility, so making this a record will not allow adding more configuration later on. What might be possible is making it an interface and using a record as an implementation.
2. I think that this class can go beyond sampler configurations. For example, wrapping is a render state, not a sampler state. Maybe something more general like "TextureParameters" can work.

Reply:
`ImageSmoothing` is very specific. This class can include many configuration options, not only from the sampler, but also from the rendered. I'm not sure what you're suggesting.

Maybe I should try to list more of the future options for this class.